posts/2021/09/hypothesis-array-api.ipynb

```diff
@@ -0,0 +1,1035 @@
+    "    a = cumulative_sums(x)\n",
+    "    # In the future NumPy should provide a public API to interop between its\n",
+    "    # ndarray and its Array API array. For now I'm using the private and\n",
+    "    # unstable x._array attribute to retrieve x's underlying ndarray.\n",
+    "    # See https://github.com/HypothesisWorks/hypothesis/issues/3101\n",
+    "    assert np.all(a._array == np.cumsum(x._array))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d409875c-4a72-42a8-9a62-36cceab5d216",
+   "metadata": {},
+   "source": [
+    "Comparison testing is a great exercise to really think about what your code does,\n",
+    "even if ultimately you conclude that you are happy with different results."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c976c705-1a8c-4308-a343-e172f72898aa",
+   "metadata": {},
+   "source": [
+    "## Watch this space\n",
+    "\n",
+    "This year should see\n",
+    "a first [version](https://data-apis.org/array-api/latest/future_API_evolution.html#versioning)\n",
+    "of the Array API standard,\n",
+    "and subsequently\n",
+    "NumPy shipping `numpy.array_api` out to the world—this means\n",
+    "array-consuming libraries will be able to\n",
+    "reliably develop for the Array API quite soon.\n",
+    "I hope I've demonstrated\n",
+    "why you should try Hypothesis\n",
+    "when the time comes 🙂\n",
+    "\n",
+    "Good news is that I'm extending my stay at Quansight.\n",
+    "My job is to help unify Python's fragmented scientific ecosystem,\n",
+    "so I'm more than happy\n",
+    "to respond to any inquiries about using Hypothesis for the Array API\n",
+    "via [email](mailto:quitesimplymatt@gmail.com) or [Twitter](https://twitter.com/whostolehonno).\n",
+    "\n",
+    "For now I'm contributing to\n",
+    "the Hypothesis-powered [Array API compliance suite](https://github.com/data-apis/array-api-tests),\n",
+    "which is already being used by the NumPy team to ensure `numpy.array_api`\n",
+    "actually complies with every tiny detail of the [specification](https://data-apis.org/array-api/latest/).\n",
+    "This process has the added side-effect\n",
+    "of finding limitations in [`hypothesis.extra.array_api`](https://hypothesis.readthedocs.io/en/latest/numpy.html#array-api),\n",
+    "so you can expect Hypothesis to only improve from here on out!"
+   ]
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "Python 3 (ipykernel)",
+   "language": "python",
+   "name": "python3"
+  },
+  "language_info": {
+   "codemirror_mode": {
+    "name": "ipython",
+    "version": 3
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.8.10"
+  },
+  "nikola": {
+   "author": "Matthew Barber",
+   "date": "2021-09-17 00:00:00 UTC",
+   "previewimage": "/images/2021/09/hypothesis-array-api-social.png",
```

Comment:
Did you on purpose not use the image in the blog post itself? Could be nice to break up the text.

Reply:
That didn't actually work, you need to use a relative link. Something like `/images/2021/...`

Reply:
Yep it won't work locally because the site preview uses the hardcoded labs.quansight.org link, not a dev environment friendly system.

Reply:
> This won't work locally because the site preview uses the hardcoded labs.quansight.org link, not a dev environment friendly system.

That's my point, it should work fine also with the dev env. All links should be `/images/....`. All other posts do it like that, why can't this link be the same?